ci(ftr): notify owners in the slack message when failures

v1v opened the pull request:
## Summary

Customise Slack messages with the Slack Team owners when the FTRs fail.

As long as the `PING_SLACK_TEAM` is set in the Buildkite pipeline, then the slack notification will poke the Slack team set in the environment variable when the step failed only.

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...





<!--ONMERGE {"backportTargets":["7.17","8.16","8.17","8.x"]} ONMERGE-->

jbudz commented on .buildkite/scripts/steps/functional/apm_cypress.sh:
```diff
@@ -29,8 +29,15 @@ fi
+  # Report the error to the team using the subfolder matching the Slack Team
+  popd
+  mkdir "$REPORT_SLACK_TEAM"
+  touch "$REPORT_SLACK_TEAM"/obs-ux-infra_services-team.slack
```
Can you help me understand the changes here vs passing a string `PING_SLACK_TEAM='obs-ux-infra_services-team`?

I'm not opposed to files but it seems to add some additional complexity.

v1v replied:
Sure thing.

I wanted to make [pingTeam](https://github.com/elastic/kibana/pull/205260/files#diff-3ca95c2f7d9589fd2954d0691e1ca0a4c2666b5ee8c6c2583242a5b97b5b253bR192) agnostic to the team name. But read the files under `target/report-slack-team`, and those files match the name of the slack team to be notified if any errors.

Regarding your question about `PING_SLACK_TEAM='obs-ux-infra_services-team'`, I'm not sure how the `post_command` can read that value and whether I could use something like `ts-node .buildkite/scripts/lifecycle/ping_team_ftrs.ts "$PING_SLACK_TEAM"` in [here](https://github.com/elastic/kibana/pull/205260/files#diff-77757ca90ebd29aff382099edd9552dd72134c1bb5ca5a2dedfe0443e0709beaR57)

Maybe I could use the `buildkite metadata set` command to notify the Slack team to reduce the additional complexity with files under `target/report-slack-team`.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/263351)
* Commit: 4f0069ccbae4381cfed2fe709826fc26c85adac4

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

cc @v1v

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/263351","number":263351,"commit":"4f0069ccbae4381cfed2fe709826fc26c85adac4"}],"number":263351}
buildkite-pr-comment-->

v1v commented on .buildkite/scripts/steps/functional/profiling_cypress.sh:
```diff
@@ -15,4 +15,4 @@ echo "--- Profiling Cypress Tests"
 NODE_OPTIONS=--openssl-legacy-provider node solutions/observability/plugins/profiling/scripts/test/e2e.js \
-  --kibana-install-dir "$KIBANA_BUILD_LOCATION" \
\ No newline at end of file
+  --kibana-install-dir "$KIBANA_BUILD_LOCATION"
```
```suggestion
  --kibana-install-dir "$KIBANA_BUILD_LOCATION" \
```

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/263385)
* Commit: d30ec5faf8f5b6af0f8fd6d8802566491ced8ab8

### Failed CI Steps
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5a-4ca6-af28-19257d37948c)
* [Jest Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5b-4c26-b6b3-18f7eb0242b2)
* [Jest Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5b-465c-9471-69beff1aefba)
* [Jest Tests #7](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5b-43e0-b169-f5735b2aadbf)
* [Jest Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5c-4ad2-b361-7db97a939adc)
* [Jest Tests #9](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5c-4366-9cb3-220e84162567)
* [Jest Tests #10](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5c-4b47-9bf1-3ed2a50184d2)
* [Jest Tests #12](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5c-46d0-8980-e7fe617ed8bc)
* [Jest Tests #13](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5d-49f6-8425-529c096b26d0)
* [Jest Tests #14](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5d-45c3-a2b8-6e8faadb034b)
* [Jest Tests #15](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5d-4e87-a88a-0a2b4327a648)
* [Jest Tests #16](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5d-4c02-b43b-19acd7c788d1)
* [Jest Tests #18](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5e-4083-a31f-744a85118731)
* [Jest Tests #19](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5e-4388-95d9-3ee2ab2c9e28)
* [Jest Tests #20](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5e-4f07-b5bb-2118261a659c)
* [Jest Integration Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5f-4f7e-89e6-d138e50f38e7)
* [Jest Integration Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5f-4adf-9316-0cfe60771647)
* [Jest Integration Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd5f-4459-aaf5-0e1db10ba9db)
* [Jest Integration Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd60-4bb7-9dee-f2d94a3a4f62)
* [Jest Integration Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd60-4ef2-a964-dcdd1c312626)
* [Jest Integration Tests #7](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd60-4a02-a718-897e627389f6)
* [FTR Configs #42](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd92-4b3e-9256-ad5444b0025b)
* [FTR Configs #48](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd96-4471-bd0d-099be9cc564c)
* [FTR Configs #52](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd99-4592-b919-e4c25a0f29b5)
* [FTR Configs #53](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd9a-4de3-9794-e0e64c66da93)
* [FTR Configs #56](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cd9d-433a-93b7-f9168281a26d)
* [FTR Configs #62](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cda0-43be-aa98-9810b5d6f689)
* [FTR Configs #63](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cda1-4627-a884-caa4cfb55eec)
* [FTR Configs #69](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cda4-45b6-b853-917beddf64c4)
* [FTR Configs #74](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cda7-4c0d-afa5-29f1d2f2fa37)
* [FTR Configs #84](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cdad-429a-9a6c-50e80a06c7ee)
* [FTR Configs #101](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cdb7-4155-8bca-9311d9f96537)
* [FTR Configs #109](https://buildkite.com/elastic/kibana-pull-request/builds/263385#01941856-cdbc-4868-a872-f023f88bdda2)

### History
* :green_heart: [Build #263351](https://buildkite.com/elastic/kibana-pull-request/builds/263351) succeeded 4f0069ccbae4381cfed2fe709826fc26c85adac4

cc @v1v

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/263385","number":263385,"commit":"d30ec5faf8f5b6af0f8fd6d8802566491ced8ab8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/263351","number":263351,"commit":"4f0069ccbae4381cfed2fe709826fc26c85adac4"}],"number":263385}
buildkite-pr-comment-->

v1v commented:
However, I've just refactored a bit more:
- [`2e28915` (#205260)](https://github.com/elastic/kibana/pull/205260/commits/2e28915247b1bc99fb0529b69ff5bc1c2f68d619)

So I created a specialised library for the Slack notifications, so I don't mix concerns with the TestFailures, in addition, I also changed the `post-command` hook to run outside of the metadata called `"${BUILDKITE_JOB_ID}_is_test_execution_step"` - as far as I see, it's set in other places but it's not set when running .buildkite/scripts/steps/functional/profiling_cypress.sh or .buildkite/scripts/steps/functional/apm_cypress.sh

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/263416)
* Commit: d2cc67c53b9a468e33d21fe74b7c7256d39cf6af

### Failed CI Steps
* [FTR Configs #41](https://buildkite.com/elastic/kibana-pull-request/builds/263416#019418b4-ef42-443a-afc5-253af8e7f862)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/263416#019418b4-ef42-443a-afc5-253af8e7f862) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/263416/jobs/019418b4-ef42-443a-afc5-253af8e7f862/artifacts/019418d7-9249-4ca7-b7f6-c35539b7e00c) FTR Configs #<span></span>41 / InfraOps App Metrics UI Node Details #<span></span>Asset Type: host Osquery Tab should show a date picker

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #263385](https://buildkite.com/elastic/kibana-pull-request/builds/263385) failed d30ec5faf8f5b6af0f8fd6d8802566491ced8ab8
* :green_heart: [Build #263351](https://buildkite.com/elastic/kibana-pull-request/builds/263351) succeeded 4f0069ccbae4381cfed2fe709826fc26c85adac4

cc @v1v

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/263416","number":263416,"commit":"d2cc67c53b9a468e33d21fe74b7c7256d39cf6af"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/263385","number":263385,"commit":"d30ec5faf8f5b6af0f8fd6d8802566491ced8ab8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/263351","number":263351,"commit":"4f0069ccbae4381cfed2fe709826fc26c85adac4"}],"number":263416}
buildkite-pr-comment-->

delanni commented on .buildkite/scripts/lifecycle/post_command.sh:
```diff
@@ -52,4 +52,12 @@ if [[ "$IS_TEST_EXECUTION_STEP" == "true" ]]; then
+if [[ $BUILDKITE_COMMAND_EXIT_STATUS -ne 0 ]]; then
+  # If the slack team environment variable is set, ping the team in slack
+  if [ -n "${PING_SLACK_TEAM:-}" ]; then
+    ts-node .buildkite/scripts/lifecycle/ping_slack.ts
```
Just a consideration: the PR is now mostly just wrapping around the typescript file, but that can be done with a simple one-liner
```
buildkite-agent meta-data set 'slack:ping_team:body' \
 "${PING_SLACK_TEAM}, can you please take a look at the test failures?"
``` 
(also add `ELASTIC_SLACK_NOTIFICATIONS_ENABLED == 'true'` above)

v1v replied:
@delanni, sure thing, I can simplify it. let me do it now

delanni replied:
Thanks! I prefer it like this, without the extra bloat for now (while this is not doing more than adding the metadata)

kibanamachine commented:
Starting backport for target branches: 7.17, 8.16, 8.17, 8.x

https://github.com/elastic/kibana/actions/runs/12651030780

kibanamachine commented:
## 💔 Some backports could not be created

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|7.17|Backport failed because of merge conflicts|
|✅|8.16|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/205715">](https://github.com/elastic/kibana/pull/205715)|
|✅|8.17|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/205716">](https://github.com/elastic/kibana/pull/205716)|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/205717">](https://github.com/elastic/kibana/pull/205717)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 205260
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
